### PR TITLE
Feature/migrate to circle ci

### PR DESCRIPTION
**Pull request by ateliercw**

### CODE REVIEW

File: Example/Pods/Pods.xcodeproj/xcshareddata/xcschemes/Swiftilities.xcscheme [hunk 1]
```diff
@@ -60,6 +79,15 @@
...
       debugDocumentVersioning = "YES">
+      <MacroExpansion>
+         <BuildableReference
+            BuildableIdentifier = "primary"
+            BlueprintIdentifier = "5776205E46583EBCE09D65E2C15E64FF"
+            BuildableName = "Swiftilities.framework"
+            BlueprintName = "Swiftilities"
+            ReferencedContainer = "container:Pods.xcodeproj">
+         </BuildableReference>
+      </MacroExpansion>
```
Comment (dostrander) [inline Example/Pods/Pods.xcodeproj/xcshareddata/xcschemes/Swiftilities.xcscheme]: #? is this from new version of cocoapods?

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+dependencies:
+  pre:
+    - brew update 1> /dev/null 2> /dev/null
+    - brew outdated carthage || brew upgrade carthage
+test:
+  override:
+    - ./scripts/iOS.sh
+  post:
+    - carthage build --no-skip-current
+    - test -d Carthage/Build/iOS/Swiftilities.framework || exit 1
```
Comment (dostrander) [inline circle.yml]: #pp `- carthage build --no-skip-current && test -d Carthage/Build/iOS/Swiftilities.framework || exit 1`

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    - bundle exec pod lib lint
+
+deployment:
+  release:
+    tag: /^\d+\.\d+\.\d+.*/
```
Comment (dostrander) [inline circle.yml]: #req I'm not sure if the travis/circle regex's are the same. But i do know this works in nima:
`/[0-9]+(\.[0-9]+)*(-.*)*/`

File: scripts/iOS.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+## Currently hardcoding the destination id while circle's OS X image is messed up
+## https://discuss.circleci.com/t/multiple-ios-simulators-available-for-single-os-device-configurations/7854
+## Once fixed replace "id=$ID" with "platform=iOS Simulator,name=iPhone 6S,OS=10.0"
+
+set -o pipefail && \
+  xcodebuild clean build test \
+  -workspace ./Example/Swiftilities.xcworkspace \
+  -scheme Swiftilities \
+  -sdk iphonesimulator \
+  -destination "id=5F273E3F-7B26-436D-A8EB-36460917614D" \
```
Comment (dostrander) [inline scripts/iOS.sh]: CAN'T WAIT TIL THIS IS FIXED!!!!! 😡

File: circle.yml (hunk 3 above)
Comment (ateliercw) [inline circle.yml]: `\d` is a digit so how about `/^\d+(\.d+)*(-.*)*/`?

**Pull request merged**